includes/Features.php

```diff
@@ -29,84 +29,110 @@ class Features {
-						'NewfoldLabs\WP\Module\Features\FeaturesCLI',
-						array(
-							'shortdesc' => 'Operations for Newfold Features.',
-							'longdesc'  => 'Internal commands to handle features.' .
-											PHP_EOL . 'Subcommands: all, enable, disable, isEnabled.',
-						)
-					);
-				}
-			);
+			add_action( 'cli_init', array( __CLASS__, 'registerCLI' ) );
+
+			// Register API script and localized values
+			add_action( 'admin_enqueue_scripts', array( __CLASS__, 'assets' ) );
 
-			// register API script
-			add_action(
-				'admin_enqueue_scripts',
-				function () {
-					$assetsDir = container()->plugin()->url . 'vendor/newfold-labs/wp-module-features/static/js/';
-					wp_enqueue_script(
-						'newfold-features',
-						$assetsDir . 'features.js',
-						array( 'wp-api-fetch' ),
-						container()->plugin()->version,
-						true
-					);
-				}
-			);
 		}
 
 		// Add default filter to make any feature null value return false
 		if ( function_exists( 'add_filter' ) ) {
 			add_filter(
 				'newfold/features/filter/isEnabled',
-				function ( $value ) {
-					// if feature state is null, return false
-					if ( ! isset( $value ) ) {
-						$value = false;
-					}
-					return $value;
-				},
+				array( __CLASS__, 'defaultIsEnabledFilter' ),
 				99, // low priority so modules can override easily if needed
 				1
 			);
 		}
 	}
 
+	/**
+	 * Find and add all feature classes to the registry
+	 */
+	public static function findFeatures() {
+		// Find extended instances of the Feature class and add to the Registry
+		foreach ( get_declared_classes() as $class ) {
+			if ( is_subclass_of( $class, 'NewfoldLabs\WP\Module\Features\Feature' ) ) {
+				// error_log( 'NewfoldLabs\WP\Module\Features child class found: '.$class );
+				// add class to registry and instantiate
+				self::$registry->set( $class );
+			}
+		}
+	}
+
+	/**
+	 * Register routes
+	 */
+	public static function registerRoutes() {
+		$api_instance = new FeaturesAPI();
+	}
+
+	/**
+	 * Add CLI commands
+	 */
+	public static function registerCLI() {
+		\WP_CLI::add_command(
+			'newfold features',
+			'NewfoldLabs\WP\Module\Features\FeaturesCLI',
+			array(
+				'shortdesc' => 'Operations for Newfold Features.',
+				'longdesc'  => 'Internal commands to handle features.' .
+								PHP_EOL . 'Subcommands: all, enable, disable, isEnabled.',
+			)
+		);
+	}
+
+	/**
+	 * Register API script and localized values
+	 */
+	public static function assets() {
+		// Register features API script
+		$scriptPath = container()->plugin()->url . 'vendor/newfold-labs/wp-module-features/static/js/features.js';
+		wp_register_script(
+			'newfold-features',
+			$scriptPath,
+			array( 'wp-api-fetch', 'newfold-features-local' ),
+			container()->plugin()->version,
+			true
+		);
+		// Register Localized Script with initial feature values
+		wp_register_script( 'newfold-features-local', null, null, container()->plugin()->version, true );
+		wp_localize_script(
+			'newfold-features-local',
```

Comment:
Can we not run `wp_localize_script()` as a dependency of `newfold-features` and do away with the pseudo `newfold-features-local` script? If it outputs after, we can just defer the script, right?